packages/tools/devtools/devtools-core/src/data-visualization/VisualSharedTreeTypes.ts

```diff
@@ -0,0 +1,58 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { type Primitive } from "./VisualTree.js";
+
+/**
+ * Base visualizer for SharedTree.
+ */
+interface SharedTreeNodeBase {
+	schema: SharedTreeSchemaNode;
+}
+
+type SharedTreeSchemaType =
```

Comment:
Verify if string union types is preferred over object constants (I think we decided not to use `enum` anymore?).